lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -594,11 +598,10 @@ def main():
                         ec2, module, domain, reuse_existing_ip_allowed,
                         module.check_mode, tag_dict, public_ipv4_pool
                     )
-                result = {
-                    'changed': changed,
-                    'public_ip': address['PublicIp'],
-                    'allocation_id': address['AllocationId']
-                }
+                result = {'changed': changed}
+                if not module.check_mode:
+                    result['public_ip'] = address['PublicIp'],
+                    result['allocation_id'] = address['AllocationId']
```

Comment:
This change appears to be non-backward compatible.

Looks like we can maintain the old result.public_ip behaviour by using this instead:

```
result.update({
    'public_ip': address['PublicIp'],
    'allocation_id': address['AllocationId']
})
```

I won't claim to be enough of a Python expert to understand why it's different...